### ISSUE TITLE
HappyPack stalls build in a docker environment

### ISSUE THREAD

**Issue by jfairley** [amireh/happypack#245]
This was 100% configuration issue on my part, but it took a while for me to figure out. Maybe this issue will help someone else in the future...

I have builds running in Docker (with CircleCI). According to CircleCI, the container has `2CPU/4096MB` available. In practice, I've seen builds stall once a node process consumes over ~1300MB. I already knew this was an issue with my Jest tests due to a different Jest / zone.js issue, but yesterday my webpack build started stalling, too.

I'm using a HappyPack thread pool as follows:

```javascript
HappyPack.ThreadPool({ size: require('os').cpus().length - 1 })
```

Eventually I noticed:

<img width="501" alt="screen shot 2018-10-09 at 12 23 31 am" src="https://user-images.githubusercontent.com/992483/46680905-5a507780-cba7-11e8-9378-9c72a24d7e1b.png">

Well yeah; that's the problem. This is the part where I make a guess, but clearly `require('os').cpus()` doesn't give the expected data inside a Docker container.

I ended up hard-coding the thread pool size when in CI environment.

```javascript
HappyPack.ThreadPool({ size: process.env.CI ? 3 : require('os').cpus().length - 1 });
```

HappyPack is happy again. 🍻

**Comment by jfairley** [comment]
@amireh Of course, nothing to do here. Close this issue when you're ready. It's merely here as documentation.

**Comment by amireh** [comment]
Ouch. Thanks for sharing that!

> This is the part where I make a guess, but clearly `require('os').cpus()` doesn't give the expected data inside a Docker container.

Personally, I wouldn't even do this even on the host machine. Upon profiling several projects on two different machines with different specs and operating systems, I've found the "happy" value to range between 3 to 5 threads (assuming the host has >= 5 virtual threads, of course.)

In theory, assigning more workers _could_ make it faster but only if the files are large enough that the overhead of IPC is justified, but I've yet to come across a project where that actually holds true.

The ["threads" option documentation](https://github.com/amireh/happypack#threads-number) briefly touches on this topic. It's stated lightly because I have no empirical evidence as it is to back any of this up. 😁 

It may be helpful to add a note of avoiding the use of `os.cpus()` in the README.
